maths/linear_diophantine_check.py

```diff
@@ -0,0 +1,101 @@
+"""
+   Program to check for solutions
+   of linear diophantine equations
+   can be possible or not
+"""
```

Comment:
Could you provide an explanation of the theorem (with sources) behind this program?